src/main/java/com/toptrumps/core/engine/Game.java

```diff
@@ -99,6 +99,11 @@ public RoundOutcome processRoundOutcome(List<Player> winners, List<Player> playe
         } else if (winners.size() == 1) {
             Player winner = winners.get(0);
             winner.setActive(true);
+
+            if(removedPlayers.contains(winner)){
+                removedPlayers.remove(winner);
+            }
```

Comment:
Great! But could you check if this would work? 

Insert the code below after ```RoundOutcome outcome```
```
  List<Player> removedPlayers = players
                .stream()
                .filter(p -> p.getDeckCount() == 0 && !winners.contains(p))
                .collect(toCollection(ArrayList::new));
```